### DISCUSSION TITLE
AttributeError: 'PySRRegressor' object has no attribute 'bumper'

### DISCUSSION THREAD

**Discussion by mrivera35** [MilesCranmer/PySR#737] [Q&A] [answered]
Getting the following error when trying to load a PySR generated pkl file:

'''
AttributeError                            Traceback (most recent call last)
Cell In[30], line 1
----> 1 flux_decline_model=pysr.PySRRegressor.from_file(model_dir+'flux_decline/flux_decline_808.pkl',model_selection='best')
      2 decay_rate_model=pysr.PySRRegressor.from_file(model_dir+'decay_rate/decay_rate_19.pkl',model_selection='best')
      3 ss_flux_model=pysr.PySRRegressor.from_file(model_dir+'ss_flux/ss_flux_691.pkl',model_selection='best')

File /opt/anaconda3/envs/pysr/lib/python3.12/site-packages/pysr/sr.py:1018, in PySRRegressor.from_file(cls, equation_file, binary_operators, unary_operators, n_features_in, feature_names_in, selection_mask, nout, **pysr_kwargs)
   1014 model.equation_file_ = os.path.join(base_dir, base_equation_file)
   1016 # Update any parameters if necessary, such as
   1017 # extra_sympy_mappings:
-> 1018 model.set_params(**pysr_kwargs)
   1019 if "equations_" not in model.__dict__ or model.equations_ is None:
   1020     model.refresh()

File /opt/anaconda3/envs/pysr/lib/python3.12/site-packages/sklearn/base.py:272, in BaseEstimator.set_params(self, **params)
    269 if not params:
    270     # Simple optimization to gain speed (inspect is slow)
    271     return self
--> 272 valid_params = self.get_params(deep=True)
    274 nested_params = defaultdict(dict)  # grouped by prefix
    275 for key, value in params.items():

File /opt/anaconda3/envs/pysr/lib/python3.12/site-packages/sklearn/base.py:244, in BaseEstimator.get_params(self, deep)
    242 out = dict()
    243 for key in self._get_param_names():
--> 244     value = getattr(self, key)
    245     if deep and hasattr(value, "get_params") and not isinstance(value, type):
    246         deep_items = value.get_params().items()

AttributeError: 'PySRRegressor' object has no attribute 'bumper'
'''

Full environment information:

'''
# Name                    Version                   Build  Channel
anyio                     4.6.2           py312hca03da5_0  
appnope                   0.1.3           py312hca03da5_1001  
argon2-cffi               21.3.0             pyhd3eb1b0_0  
argon2-cffi-bindings      21.2.0          py312h80987f9_0  
asttokens                 2.0.5              pyhd3eb1b0_0  
async-lru                 2.0.4           py312hca03da5_0  
attrs                     24.2.0          py312hca03da5_0  
babel                     2.11.0          py312hca03da5_0  
beautifulsoup4            4.12.3          py312hca03da5_0  
bleach                    4.1.0              pyhd3eb1b0_0  
brotli                    1.0.9                h80987f9_8  
brotli-bin                1.0.9                h80987f9_8  
brotli-python             1.0.9           py312h313beb8_8  
bzip2                     1.0.8                h80987f9_6  
ca-certificates           2024.9.24            hca03da5_0  
certifi                   2024.8.30       py312hca03da5_0  
cffi                      1.17.1          py312h3eb5a62_0  
charset-normalizer        3.3.2              pyhd3eb1b0_0  
click                     8.1.7           py312hca03da5_0  
comm                      0.2.1           py312hca03da5_0  
contourpy                 1.3.0           py312h6142ec9_2    conda-forge
cycler                    0.11.0             pyhd3eb1b0_0  
debugpy                   1.6.7           py312h313beb8_0  
decorator                 5.1.1              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
entrypoints               0.4             py312hca03da5_0  
et_xmlfile                1.1.0           py312hca03da5_1  
executing                 0.8.3              pyhd3eb1b0_0  
expat                     2.6.3                h313beb8_0  
fonttools                 4.51.0          py312h80987f9_0  
freetype                  2.12.1               hadb7bae_2    conda-forge
glib                      2.82.2               hb1db9eb_0    conda-forge
glib-tools                2.82.2               h25d4a46_0    conda-forge
glob2                     0.7                        py_0    conda-forge
gst-plugins-base          1.24.7               hb49d354_0    conda-forge
gstreamer                 1.24.7               hc3f5269_0    conda-forge
h11                       0.14.0          py312hca03da5_0  
httpcore                  1.0.2           py312hca03da5_0  
httpx                     0.27.0          py312hca03da5_0  
icu                       73.2                 hc8870d7_0    conda-forge
idna                      3.7             py312hca03da5_0  
ipykernel                 6.29.5          py312hca03da5_0  
ipython                   8.27.0          py312hca03da5_0  
ipywidgets                8.1.2           py312hca03da5_0  
jedi                      0.19.1          py312hca03da5_0  
jinja2                    3.1.4           py312hca03da5_0  
joblib                    1.4.2           py312hca03da5_0  
json5                     0.9.6              pyhd3eb1b0_0  
jsonschema                4.23.0          py312hca03da5_0  
jsonschema-specifications 2023.7.1        py312hca03da5_0  
jupyter                   1.1.1              pyhd8ed1ab_0    conda-forge
jupyter-lsp               2.2.0           py312hca03da5_0  
jupyter_client            8.6.0           py312hca03da5_0  
jupyter_console           6.6.3           py312hca03da5_1  
jupyter_core              5.7.2           py312hca03da5_0  
jupyter_events            0.10.0          py312hca03da5_0  
jupyter_server            2.14.1          py312hca03da5_0  
jupyter_server_terminals  0.4.4           py312hca03da5_1  
jupyterlab                4.2.5           py312hca03da5_0  
jupyterlab_pygments       0.1.2                      py_0  
jupyterlab_server         2.27.3          py312hca03da5_0  
jupyterlab_widgets        3.0.10          py312hca03da5_0  
kiwisolver                1.4.4           py312h313beb8_0  
krb5                      1.21.3               hf3e1bf2_0  
lcms2                     2.16                 ha0e7c42_0    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libblas                   3.9.0           1_h9886b1c_netlib  
libbrotlicommon           1.0.9                h80987f9_8  
libbrotlidec              1.0.9                h80987f9_8  
libbrotlienc              1.0.9                h80987f9_8  
libcblas                  3.9.0           1_h9886b1c_netlib  
libclang-cpp15            15.0.7          default_he012953_5    conda-forge
libclang13                19.1.2          default_h5f28f6d_1    conda-forge
libcxx                    19.1.2               ha82da77_0    conda-forge
libdeflate                1.22                 hd74edd7_0    conda-forge
libedit                   3.1.20230828         h80987f9_0  
libexpat                  2.6.3                hf9b8971_0    conda-forge
libffi                    3.4.4                hca03da5_1  
libgfortran               5.0.0           13_2_0_hd922786_3    conda-forge
libgfortran5              13.2.0               hf226fd6_3    conda-forge
libglib                   2.82.2               h07bd6cf_0    conda-forge
libiconv                  1.17                 h0d3ecfb_2    conda-forge
libintl                   0.22.5               h8414b35_3    conda-forge
libintl-devel             0.22.5               h8414b35_3    conda-forge
libjpeg-turbo             3.0.3                h80987f9_0  
liblapack                 3.9.0           1_h9886b1c_netlib  
libllvm15                 15.0.7               h2621b3d_4    conda-forge
libllvm19                 19.1.2               haf57ff0_0    conda-forge
libogg                    1.3.5                h1a28f6b_1  
libopus                   1.3.1                h80987f9_1  
libpng                    1.6.44               hc14010f_0    conda-forge
libpq                     16.4                 hfb0b52a_3    conda-forge
libsodium                 1.0.18               h1a28f6b_0  
libsqlite                 3.45.2               h091b4b1_0    conda-forge
libtiff                   4.7.0                hfce79cd_1    conda-forge
libvorbis                 1.3.7                h1a28f6b_0  
libwebp-base              1.4.0                h93a5062_0    conda-forge
libxcb                    1.17.0               hdb1d25a_0    conda-forge
libxml2                   2.12.7               h9a80f22_3    conda-forge
libzlib                   1.3.1                h8359307_2    conda-forge
llvm-openmp               19.1.2               hb52a8e5_0    conda-forge
markupsafe                2.1.3           py312h80987f9_0  
matplotlib                3.9.2           py312h1f38498_1    conda-forge
matplotlib-base           3.9.2           py312h9bd0bc6_1    conda-forge
matplotlib-inline         0.1.6           py312hca03da5_0  
mistune                   2.0.4           py312hca03da5_0  
mpmath                    1.3.0           py312hca03da5_0  
mysql-common              8.3.0                h1687695_5    conda-forge
mysql-libs                8.3.0                h0e80b4a_5    conda-forge
nbclient                  0.8.0           py312hca03da5_0  
nbconvert                 7.16.4               hd8ed1ab_1    conda-forge
nbconvert-core            7.16.4             pyhd8ed1ab_1    conda-forge
nbconvert-pandoc          7.16.4               hd8ed1ab_1    conda-forge
nbformat                  5.10.4          py312hca03da5_0  
ncurses                   6.4                  h313beb8_0  
nest-asyncio              1.6.0           py312hca03da5_0  
notebook                  7.2.2           py312hca03da5_1  
notebook-shim             0.2.3           py312hca03da5_0  
nspr                      4.35                 h313beb8_0  
nss                       3.98                 h5ce2875_0    conda-forge
numpy                     2.1.2           py312h801f5e3_0    conda-forge
openjpeg                  2.5.2                h54b8e55_0  
openpyxl                  3.1.5           py312h80987f9_0    anaconda
openssl                   3.3.2                h8359307_0    conda-forge
overrides                 7.4.0           py312hca03da5_0  
packaging                 24.1            py312hca03da5_0  
pandas                    2.2.3           py312hcd31e36_1    conda-forge
pandoc                    2.12                 hca03da5_3  
pandocfilters             1.5.0              pyhd3eb1b0_0  
parso                     0.8.3              pyhd3eb1b0_0  
pcre2                     10.44                h297a79d_2    conda-forge
pexpect                   4.8.0              pyhd3eb1b0_3  
pillow                    11.0.0          py312haf37ca6_0    conda-forge
pip                       24.2            py312hca03da5_0  
platformdirs              3.10.0          py312hca03da5_0  
ply                       3.11            py312hca03da5_1  
prometheus_client         0.14.1          py312hca03da5_0  
prompt-toolkit            3.0.43          py312hca03da5_0  
prompt_toolkit            3.0.43               hd3eb1b0_0  
psutil                    5.9.0           py312h80987f9_0  
pthread-stubs             0.3                  h1a28f6b_1  
ptyprocess                0.7.0              pyhd3eb1b0_2  
pure_eval                 0.2.2              pyhd3eb1b0_0  
pycparser                 2.21               pyhd3eb1b0_0  
pygments                  2.15.1          py312hca03da5_1  
pyjuliacall               0.9.23             pyhd8ed1ab_0    conda-forge
pyjuliapkg                0.1.13             pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.2           py312hca03da5_0  
pyqt                      5.15.10         py312h313beb8_0    anaconda
pyqt5-sip                 12.13.0         py312h80987f9_0  
pyqt6                     6.7.1                    pypi_0    pypi
pyqt6-qt6                 6.7.3                    pypi_0    pypi
pyqt6-sip                 13.8.0                   pypi_0    pypi
pysocks                   1.7.1           py312hca03da5_0  
pysr                      0.19.4             pyhd8ed1ab_0    conda-forge
python                    3.12.2          hdf0ec26_0_cpython    conda-forge
python-dateutil           2.9.0post0      py312hca03da5_2  
python-fastjsonschema     2.16.2          py312hca03da5_0  
python-json-logger        2.0.7           py312hca03da5_0  
python-tzdata             2023.3             pyhd3eb1b0_0  
python_abi                3.12                    5_cp312    conda-forge
pytz                      2024.1          py312hca03da5_0  
pyyaml                    6.0.2           py312h80987f9_0  
pyzmq                     25.1.2          py312h313beb8_0  
qhull                     2020.2               h48ca7d4_2  
qt-main                   5.15.8              hf679f28_21    conda-forge
readline                  8.2                  h1a28f6b_0  
referencing               0.30.2          py312hca03da5_0  
requests                  2.32.3          py312hca03da5_0  
rfc3339-validator         0.1.4           py312hca03da5_0  
rfc3986-validator         0.1.1           py312hca03da5_0  
rpds-py                   0.10.6          py312hf0e4da2_0  
scikit-learn              1.5.2           py312h387f99c_1    conda-forge
scipy                     1.14.1          py312heb3a901_0    conda-forge
semver                    3.0.2           py312hca03da5_0  
send2trash                1.8.2           py312hca03da5_0  
setuptools                75.1.0          py312hca03da5_0  
sip                       6.7.12          py312h313beb8_0  
six                       1.16.0             pyhd3eb1b0_1  
sniffio                   1.3.0           py312hca03da5_0  
soupsieve                 2.5             py312hca03da5_0  
sqlite                    3.45.2               hf2abe2d_0    conda-forge
stack_data                0.2.0              pyhd3eb1b0_0  
sympy                     1.13.2          py312hca03da5_0  
terminado                 0.17.1          py312hca03da5_0  
threadpoolctl             3.5.0           py312h989b03a_0  
tinycss2                  1.2.1           py312hca03da5_0  
tk                        8.6.13               h5083fa2_1    conda-forge
tornado                   6.4.1           py312h80987f9_0  
traitlets                 5.14.3          py312hca03da5_0  
typing-extensions         4.11.0          py312hca03da5_0  
typing_extensions         4.11.0          py312hca03da5_0  
tzdata                    2024b                h04d1e81_0  
unicodedata2              15.1.0          py312h80987f9_0  
urllib3                   2.2.3           py312hca03da5_0  
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1           py312hca03da5_2  
websocket-client          1.8.0           py312hca03da5_0  
wheel                     0.44.0          py312hca03da5_0  
widgetsnbextension        4.0.10          py312hca03da5_0  
xorg-libxau               1.0.11               hd74edd7_1    conda-forge
xorg-libxdmcp             1.1.5                hd74edd7_0    conda-forge
xz                        5.4.6                h80987f9_1  
yaml                      0.2.5                h1a28f6b_0  
zeromq                    4.3.5                h313beb8_0  
zlib                      1.3.1                h8359307_2    conda-forge
zstd                      1.5.6                hb46c0d2_0    conda-forge
'''

**Comment by MilesCranmer** [comment, marked as answer]
Maybe you saved on an older version of PySR and attempting to load it on the latest? Or the reverse?

**Reply by mrivera35** [reply]
Yes, I was attempting to load models that were generated on an older version. I regressed the version and that fixed the issue. Thank you!
